pkg/distributor/distributor.go

```diff
@@ -749,7 +763,9 @@ func (d *Distributor) prePushHaDedupeMiddleware(next push.Func) push.Func {
 			}
 
 			if errors.Is(err, tooManyClustersError{}) {
-				d.discardedSamplesTooManyHaClusters.WithLabelValues(userID).Add(float64(numSamples))
+				group := validation.GroupLabel(d.limits, userID, req.Timeseries)
+
```

Comment:
We should update the active group timestamp (and apply limits) here too. I'm concerned about this: if you missed this in the original PR, someone may miss it in the future PRs and leak a group label on the exposed metric.